### PR TITLE
Change yaml marshalling of node-labels

### PR DESCRIPTION
**Pull request by frelon**
The yaml marshalling does not play well with '+' in the key.

This commit makes the node-labels hopefully play nice with k3s/rke config again.

Signed-off-by: Fredrik Lönnegren <fredrik.lonnegren@suse.com>

### CODE REVIEW

**Review by alexander-demicev — COMMENTED** [review]
The PR looks good to me, should the fix go directly to the new branch or wait until we merge the new operator in the main?

**Review by kkaempf — CHANGES_REQUESTED** [review]
Shouldn't all of this be a `git revert` instead ?

**Comment by frelon** [comment]
> Shouldn't all of this be a `git revert` instead ?

Well, with the + in the yaml tag it will not serialize properly so we still need the ugly work-around.

File: pkg/controllers/machineinventoryselector/bootstrap.go
Comment (frelon) [reply pkg/controllers/machineinventoryselector/bootstrap.go:236]: I mentioned this in the comment for the function, but can add a comment here as well!

**Comment by frelon** [comment]
> The PR looks good to me, should the fix go directly to the new branch or wait until we merge the new operator in the main?

Probably easiest to go to main directly? Unless you plan to merge the new operator today?

**Review by Itxaka — CHANGES_REQUESTED** [review]
Dont agree with this solutiuon. We should not force the yaml into a string wheh we have libs for that.

The real issue here is that the k8s yaml library FOR SOME WEIRD REASON, uses the json comment in the struct to know how to marshall the end yaml. dont ask me why it doesnt use the EXPLICIT YAML FIELD for that but it doesnt.

Just do a revert of the old commit and change the struct to use json comment instead of yaml and you will see it working back. Baffling.

See https://go.dev/play/p/GnYFrZVwfIn for an example and check the different in output.

So we need to add:

```
type LabelsFromInventory struct {
		NodeLabels []string `yaml:"node-label+",json:"node-label+"`
	}
```

**Comment by frelon** [comment]
> Dont agree with this solutiuon. We should not force the yaml into a string wheh we have libs for that.
> 
> The real issue here is that the k8s yaml library FOR SOME WEIRD REASON, uses the json comment in the struct to know how to marshall the end yaml. dont ask me why it doesnt use the EXPLICIT YAML FIELD for that but it doesnt.
> 
> Just do a revert of the old commit and change the struct to use json comment instead of yaml and you will see it working back. Baffling.
> 
> See https://go.dev/play/p/GnYFrZVwfIn for an example and check the different in output.
> 
> So we need to add:
> 
> ```
> type LabelsFromInventory struct {
> 		NodeLabels []string `yaml:"node-label+",json:"node-label+"`
> 	}
> ```

Speechless.. But yes that looks much better! I will make the change!

**Comment by frelon** [comment]
Hmm, looking at the code again I don't know why we are using _sigs.k8s.io/yaml_ as we use _gopkg.in/yaml.v2_ in other parts of the code. Will change to use that one instead!

https://go.dev/play/p/vGEq1fqUelY

**Review by Itxaka — APPROVED** [review]

**Comment by Itxaka** [comment]
weird as hell

**Pull request merged**
